src/parser.rs

```diff
@@ -423,6 +424,74 @@ impl Parser {
         })
     }
 
+    /// Parse a SQL LISTAGG expression, e.g. `LISTAGG(...) WITHIN GROUP (ORDER BY ...)`.
+    pub fn parse_listagg_expr(&mut self) -> Result<Expr, ParserError> {
+        self.expect_token(&Token::LParen)?;
+        let all = self.parse_keyword("ALL");
```

Comment:
I've gone ahead and done this but my first pass of this generalizes the error message in a way that removes preexisting context. I completely understand if this isn't desirable and would be happy to adapt it so that information isn't lost. On the other hand, I wasn't necessarily sure it was needed. So, I'll defer to you and then update accordingly.

Reply:
I wouldn't worry too much about the error message. Good error messages should provide location information, not explain it with words.